index.js

```diff
@@ -9,14 +9,23 @@ function search(dir) {
   var context = path.resolve(dir)
   if (!fs.existsSync(context)) return []
   var contents = fs.readdirSync(context)
-  return contents.map(function(name) {
+  var scopedRoots = []
+  var results = contents.filter(function (name) {
+    var isBin = name === ".bin"
+    var isScoped = name.charAt(0) === "@"
+    if (isScoped) scopedRoots.push(path.join(dir, name))
+    return !isScoped && !isBin
+  }).map(function(name) {
     var relative = path.join(dir, name)
     var found = new Found
     found.name = name
     found.path = path.resolve(relative)
     if (is(relative)) found.link = read(relative)
     return found
   })
+  return scopedRoots.reduce(function (_, scopedRoot) {
+    return _.concat(search(scopedRoot))
+  }, results)
```

Comment:
I thought this was lodash at first so if reducing let's avoid `_` as the accumulation variable name. I actually wonder if there is another technique we can use above that would avoid the need to reduce. Is the scoped solution to just search deep or is there more to it than that?

Reply:
So the filter step above was added to remove processing on scoped directories. For these, it needs to instead look up all their sub-directories (packages) and run it for them. This produces a multi-dimensional array.

Main point of the reduce is to flatten the arrays. Running `search` on each of the `scopedRoots` directories produces a new array of results.

```
@foo/
    a/
    b/
@bar
    c/
d/
e/
```

**scopedRoots**
```json
[ "@foo", "@bar" ]
```

```js
const scopedResults = scopedRoots.map(search);
```

**scopedResults**
```json
[
    [ "@foo/a", "@foo/b" ],
    [ "@foo/c" ]
]
```

This would then need an extra step to reduce it to:
```json
[
    "@foo/a",
    "@foo/b",
    "@foo/c"
```

So instead of all that, I used reduce to flatten as it searches each scoped directory. I'm sure there are other ways to rewrite it, but that's the gist of it.

Reply:
@ryanve thinking about it more, I think the entire `filter` / `map` / `reduce` would be better as a single `reduce`. I'll kick something up in next commit.